### PR TITLE
Abstract revert functionality

### PR DESCRIPTION
**Pull request by dometto**
Simplify the implementation of `Wiki#revert_page` by abstracting low-level grit calls to the grit adapter (https://github.com/gollum/grit_adapter/pull/14). I've manually tested that this branch with the corresponding branch of `grit_adapter` behaves the same as `master`, but would appreciate if someone else could verify this (since the revert tests don't work with grit).

### CODE REVIEW

**Comment by dometto** [comment]
Note: to test this you have to manually add the following to `Gemfile`:

```ruby
gem `gollum-grit_adapter`, :git => 'https://github.com/gollum/grit_adapter/', :branch => 'abstract_revert'
```

And remove the `grit_adapter` dependency from `gemspec.rb`.

File: lib/gollum-lib/wiki.rb [hunk 1]
```diff
@@ -510,10 +510,8 @@ def revert_page(page, sha1, sha2 = nil, commit = {})
         sha2   = nil
       end
 
-      patch     = full_reverse_diff_for(page, sha1, sha2)
       committer = Committer.new(self, commit)
-      parent    = committer.parents[0]
-      committer.options[:tree] = @repo.git.apply_patch(parent.sha, patch)
+      committer.options[:tree] = @repo.git.revert(page ? page.path : nil, sha1, sha2, committer.parents[0])
```
Comment (bartkamphorst) [inline lib/gollum-lib/wiki.rb]: I'm not sure I understand this behavior here, as the method is called `revert_page`. So I would assume that this method has to be called with page object. There is a separate `revert_commit` method below, which should do something else than revert_page.

Comment (dometto) [inline lib/gollum-lib/wiki.rb]: Yes, but `revert_commit` just calls `revert_page` with `nil` as the `page` argument. If we're changing the API anyway we can try to make this less counterintuitive. What about these three methods:
* private: `revert(page = nil, sha1, sha2, commit = {})` (implements current logic of `revert_page`)
* public: `revert_page(page, sha1, sha2, commit = {})` (calls `revert` with a page argument)
* public: `revert_commit(sha1, sha2, commit = {})` (calls revert_commit)

**Comment by dometto** [comment]
Refactred according to your suggestions. Would be great if you could do a manual test to verify this behaves the same as the current version of `gollum-lib`.

**Comment by dometto** [comment]
Also, I guess this is a breaking change. Should I reopen this against a new `5.x` branch?

File: lib/gollum-lib/wiki.rb [hunk 2]
```diff
@@ -922,5 +857,62 @@ def tree_map_for(ref, ignore_page_file_dir=false)
     def inspect
       %(#<#{self.class.name}:#{object_id} #{@repo.path}>)
     end
+
+    private
+
+    # Public: Applies a reverse diff for a given page, or an entire commit if page is nil. If only 1 SHA is given,
```
Comment (bartkamphorst) [inline lib/gollum-lib/wiki.rb]: Is this still Public?

**Comment by bartkamphorst** [comment]
> Also, I guess this is a breaking change. Should I reopen this against a new 5.x branch?

That would be good, yes.

File: lib/gollum-lib/wiki.rb (hunk 2 above)
Comment (dometto) [inline lib/gollum-lib/wiki.rb]: Ah no, thanks.

**Comment by bartkamphorst** [comment]
Okay, I just tested out the `simplify_revert` branch with the `abstract_revert` grit adapter and reverting pages works without a hitch. :+1:

**Pull request closed without merging**